app/models/concerns/host_common.rb

```diff
@@ -151,30 +151,29 @@ def image_file
     super || default_image_file
   end
 
-  def crypt_root_pass
-    # hosts will always copy and crypt the password from parents when saved, but hostgroups should
-    # only crypt if the attribute is stored, else will stay blank and inherit
-    unencrypted_pass = if is_a?(Hostgroup)
-                         self[:root_pass]
-                       else
-                         root_pass
-                       end
-
-    if unencrypted_pass.present?
-      is_actually_encrypted = if (operatingsystem.try(:password_hash) == "Base64" || operatingsystem.try(:password_hash) == "Base64-Windows")
-                                password_base64_encrypted?
-                              elsif PasswordCrypt.crypt_gnu_compatible?
-                                unencrypted_pass.match('^\$\d+\$.+\$.+')
-                              else
-                                unencrypted_pass.starts_with?("$")
-                              end
-
-      if is_actually_encrypted
-        self.root_pass = self.grub_pass = unencrypted_pass
-      else
-        self.root_pass = operatingsystem.nil? ? PasswordCrypt.passw_crypt(unencrypted_pass) : PasswordCrypt.passw_crypt(unencrypted_pass, operatingsystem.password_hash)
-        self.grub_pass = PasswordCrypt.grub2_passw_crypt(unencrypted_pass)
-      end
+  def save_crypted_root_pass
+    self.root_pass = crypt_pass(self[:root_pass], :root)
+    self.grub_pass = crypt_pass(self[:grub_pass] || self[:root_pass], :grub)
+  end
+
+  def crypt_pass(unencrypted_pass, pass_kind)
+    return unless unencrypted_pass.present?
+    raise "Correct type of password is not used. You can use only :root or :grub" if [:root, :grub].exclude?(pass_kind)
```

Comment:
It is bit hard to read and decode what does it mean.
```suggestion
    raise "Incorrect type of password. Only one of :root, :grub is supported" unless [:root, :grub].include?(pass_kind)
```